x-pack/plugins/security_solution/public/common/components/visualization_actions/use_lens_attributes.tsx

```diff
@@ -40,8 +41,11 @@ export const useLensAttributes = ({
   const [{ detailName, pageName, tabName }] = useRouteSpy();
 
   const tabsFilters = useMemo(() => {
```

Comment:
Agree Pablo, we could probably get those `tabsFilters` and `pageFilters` in a grouped `filters` parameter, passed from the components, that know the filter they are applying, instead of checking in which page we are here to add one thing or the other. We shouldn't have to remember to come here when we change other parts.

I am also wondering why we need the `getLensAttributes` and `stackByField` parameters, we could just use `lensAttributes` by just calling the `getLensAttributes(stackByField)` outside this hook.

I think we should change these couple of things if it is possible.
cc @stephmilovic